src/pycsou/opt/solver/fw_lasso.py

```diff
@@ -0,0 +1,463 @@
+
+    ``VanillaFWforLasso.fit()`` **Parametrisation**
+
+    compute_ofv: Bool
+        Indicator to keep track of the value of the objective function along the iterations, default is True.
+        This value is optional for running the Vanilla FW iterations, but can be used to determine a stopping criterion.
+    """
+
+    def __init__(
+        self,
+        data: pyct.NDArray,
+        forwardOp: pyco.LinOp,
+        lambda_: float,
+        step_size: str = "optimal",
+        *,
+        folder: typ.Optional[pyct.PathLike] = None,
+        exist_ok: bool = False,
+        writeback_rate: typ.Optional[int] = None,
+        verbosity: int = 50,
+        show_progress: bool = True,
+        log_var: pyct.VarName = (
+            "x",
+            "dcv",
+        ),
+    ):
+        r"""
+
+        Parameters
+        ----------
+        data: NDArray
+            (..., L) measurements term(s) in the LASSO objective function. The type of NDArray defines the module used
+            for the iterates.
+        forwardOp: LinOp
+            (N, L) measurement operator in the LASSO objective function.
+        lambda_: float
+            Regularisation parameter from the LASSO problem.
+        step_size: str
+            ["regular", "optimal"] If ``step_size="regular"``, the convex combination weight in the weighting step of
+            FW is set according to the default strategy, :math:`\gamma_k = 2/k+2`.
+            If ``step_size="regular"``, the convex combination weight :math:`\gamma_k` is computed at each step
+            as the optimal step size, for which there exists a close form formulation in the specific case of minimising
+            the LASSO objective function.
+        """
+        super().__init__(
+            data=data,
+            forwardOp=forwardOp,
+            lambda_=lambda_,
+            folder=folder,
+            exist_ok=exist_ok,
+            writeback_rate=writeback_rate,
+            verbosity=verbosity,
+            show_progress=show_progress,
+            log_var=log_var,
+        )
+
+        try:
+            assert step_size in ["optimal", "regular"]
+            self.step_size_strategy = step_size
+        except:
+            raise ValueError(f"step_size must be in ['regular', 'optimal'], got {step_size}.")
+
+    def m_init(self, **kwargs):
+        super(VanillaFWforLasso, self).m_init(**kwargs)
+        self._mstate["lift_variable"] = 0.0  # todo [0]
+        # QfR: Do I need to enforce the precision on plain floats ?
+
+    def m_step(self):
+        mst = self._mstate  # shorthand
+        xp = pycu.get_array_module(mst["x"])
+        mgrad = -self._data_fidelity.grad(mst["x"])
+        new_ind = xp.argmax(xp.abs(mgrad), axis=-1)
+        dcv = mgrad[new_ind] / self.lambda_  # todo [0]
+        mst["dcv"] = dcv
+        # To be careful about mst["dcv"] as it can be either float if new_ind is size 1 or NDArray if new_ind is size>1
```

Comment:
I am not sure if this is an issue, but in case you wanted to always have NDArray you could use: 
```
new_ind = xp.empty(mgrad.shape[:-1], dtype=int32)
xp.argmax(xp.abs(mgrad), axis=-1, out=new_ind)
```